File: Codeforces/div821/A.cpp

#include <iostream>
#include <vector>
using namespace::std;
int main()
{
	int t;
	cin >> t;
	while (t--)
	{
		int n, k;
		cin >> n >> k;
		vector<int> a(n);
		for (int i=0; i < n; ++i)
			cin >> a.at(i);
		long long ans = 0;
		for (int i=0; i < k; ++i)
		{
			int m = 0;
			for (int j=i; j < n; j += k)
				m = max(m, a[j]);
			ans += m;
		}
		cout << ans << endl;
	}
}

File: Codeforces/div821/A.py

def inp():
    return [int(i) for i in input().split()]

for _ in range(int(input())):
    n, k = inp()
    a = inp()
    ans = 0
    for i in range(k):
        ans += max(a[i] for i in range(i, n, k))
    print(ans)

